x-pack/plugins/apm/public/context/UrlParamsContext/index.tsx

```diff
@@ -27,6 +28,16 @@ interface TimeRangeRefreshAction {
   time: TimeRange;
 }
 
+function getUIFilters(urlParams: IUrlParams): UIFilters {
+  return useMemo(
+    () => ({
+      kuery: urlParams.kuery,
+      environment: urlParams.environment
+    }),
+    [urlParams]
```

Comment:
This is currently being recomputed too often. Whenever a `urlParam` is changed it will also cause `uiFilters` to update which in turn will cause `useFetcher` to load data.

This can be seen on the list of services where it is currently loading `services` twice. This is also happening on other pages.


I think this will solve it (but please verify)
```ts
function getUIFilters({ kuery, environment }: IUrlParams): UIFilters {
  return useMemo(
    () => ({
      kuery,
      environment
    }),
    [kuery, environment]
  );
}
```